src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.NetCoreApp.cs

```diff
@@ -4,16 +4,29 @@
 
 using System;
 using System.Diagnostics;
+#if NETCOREAPP
+using System.Buffers.Binary;
+#endif
 
 namespace Microsoft.Data.SqlClient
 {
     internal sealed partial class TdsParser
     {
         internal static void FillGuidBytes(Guid guid, Span<byte> buffer) => guid.TryWriteBytes(buffer);
 
-        internal static void FillDoubleBytes(double value, Span<byte> buffer) => BitConverter.TryWriteBytes(buffer, value);
+        internal static void FillDoubleBytes(double value, Span<byte> buffer) => 
+#if NETCOREAPP
+        BinaryPrimitives.WriteDoubleLittleEndian(buffer, value);
```

Comment:
Is WriteDoubleBigEndian available in all build targets? The [documentation](https://learn.microsoft.com/en-us/dotnet/api/system.buffers.binary.binaryprimitives.writedoublebigendian?view=net-8.0) says it is supported since .NET 5. Same as ReadDoubleBigEndian, which you are not using.

You also used WriteSingleLittleEndian in the next few lines.

Reply:
Exactly what I was thinking. I didn't try to compile it to verify there is an error, but it looks like it.